src/FeatureChecker.ts

```diff
@@ -42,6 +41,42 @@ export default class FeatureChecker {
     if (this.#crash) process.exit(1);
   }
 
+  public async checkGuild(guild: Guild): Promise<void> {
+    const guildInfo: string = `for guild ${guild.id} (${guild.name})`;
+    let error: boolean = false;
+    try {
+      const info: GuildInfo = await GuildSettings.settings(guild.id);
+      if (info.permissionRoles.length === 0)
+        this.status(`No permitted roles set for ${guildInfo}`);
+      if (info.roles.length === 0) this.status(`Roles disabled ${guildInfo}`);
+      else this.status(`Roles enabled ${guildInfo}`);
+      if (info.roles.length > 25) {
+        error = true;
+        this.error(
+          `Currently, only a maximum of 25 roles is allowed. Guild ${guild.id} (${guild.name}) has more than 25 roles. If you need more, file an issue at https://github.com/TrojanerHD/TrojanerBot/issues/new`,
+          false
+        );
+      }
+    } catch (e: unknown) {
+      error = true;
+      this.error(
+        `Could not load information ${guildInfo} with reason ${e}`,
+        false
+      );
+    } finally {
+      if (!error) {
+        DiscordClient._safeGuilds.push(guild);
+        if (!GuildSettings.settings(guild.id))
+          await GuildSettings.saveSettings(guild, {
+            permissionRoles: [],
+            roles: [],
+            streamers: [],
+            refreshToken: '',
+          }).catch(console.error);
+      } else this.warning(`Limited features due to problems ${guildInfo}`);
```

Comment:
Remove a guild from the safe guilds if there was a problem to make this function safe when re-executing